libs/azure-storage/pyproject.toml

```diff
@@ -67,6 +67,7 @@ optional = true
 
 [tool.poetry.group.test.dependencies]
 pytest = "~7.4"
+pytest-mock = "~3.14"
```

Comment:
Let's hold off pulling in `pytest-mock` and instead let's just use the `mock` module from `unittest`. Mainly I'm not sure if there is a strong reason we need this pytest plugin since `pytest-mock` is a slim wrapper over `mock` and it's advantageous to keep our dependency list as lean as possible.